addons/io_hubs_addon/components/operators.py

```diff
@@ -670,6 +671,273 @@ def invoke(self, context, event):
+                self.render_file_format_tmp = context.scene.render.image_settings.file_format
+                context.scene.render.engine = 'CYCLES'
+                self.cycles_samples_tmp = context.scene.cycles.samples
+                context.scene.cycles.samples = self.samples
+                # Baking needs to happen without the color pass because we only want the direct and indirect light contributions
+                self.bake_settings_before = context.scene.render.bake
+                bake_settings = context.scene.render.bake
+                bake_settings.use_pass_direct = True
+                bake_settings.use_pass_indirect = True
+                bake_settings.use_pass_color = False
+                # The should be small because otherwise it could overwrite UV islands
+                bake_settings.margin = 2
+                # Get the image type from the enum.
+                # Because it stores only the index we need all items first and then select by index.
+                enum_items = self.properties.bl_rna.properties['image_type'].enum_items
+                self.image_type_name = enum_items.get(self.image_type).name
+                # Not sure whether this has any influence
+                bake_settings.image_settings.file_format = self.image_type_name
+                context.scene.render.image_settings.file_format = self.image_type_name
+
+                bpy.ops.object.bake('INVOKE_DEFAULT', type='DIFFUSE')
+                self.bake_started = True
+                return {"PASS_THROUGH"}
+            elif bpy.app.is_job_running('OBJECT_BAKE'):
+                return {"PASS_THROUGH"}
+            else:
+                # Pack all newly created or updated images
+                for node in self.lightmap_texture_nodes:
+                    file_path = bpy.path.abspath(f"{bpy.app.tempdir}/{node.image.name}.{self.image_type_name.lower()}")
+                    # node.image.save_render(file_path)
+                    node.image.filepath_raw = file_path
+                    node.image.file_format = self.image_type_name
+                    node.image.save()
+                    node.image.pack()
+                    # Update the filepath so it unpacks nicely for the user.
+                    # TODO: Mechanism taken from reflection_probe.py line 300-306, de-duplicate
+                    new_filepath = f"//{node.image.name}.{self.image_type_name.lower()}"
+                    node.image.packed_files[0].filepath = new_filepath
+                    node.image.filepath_raw = new_filepath
+                    node.image.filepath = new_filepath
+
+                    # Remove file from temporary directory to de-clutter the system. Especially on windows the temporary directory is rarely purged.
+                    if os.path.exists(file_path):
+                        os.remove(file_path)
+
+                # return to old settings
+                bake_settings = self.bake_settings_before
```

Comment:
This doesn't restore the old bake settings, they need to be handled one by one, the same as the render settings.